src/js/components/dropdown.js

```diff
@@ -0,0 +1,286 @@
+    menuList.setAttribute('aria-hidden', 'false');
+
+    this.activeDropdown = this.element;
+  }
+
+  /**
+   * Closes a dropdown menu
+   * @param {HTMLButtonElement} toggle - a button element which toggles the dropdown
+   * @param {HTMLDivElement} menu - a div containing the dropdown menu items
+   */
+  closeMenu(toggle, menu) {
+    const toggleButton = toggle || this.toggleElement;
+    const menuList = menu || this.menuElement;
+
+    // Return if disabled dropdown is being closed programmatically
+    if (toggleButton.hasAttribute('disabled')) {
+      return;
+    }
+
+    if (toggleButton && toggleButton !== undefined) {
+      toggleButton.setAttribute('aria-expanded', 'false');
+    }
+
+    menuList.setAttribute('aria-hidden', 'true');
+  }
+
+  /**
+   * Creates an object of menu items within a dropdown
+   * @param {HTMLDivElement} menu - a div containing the dropdown menu items
+   * @returns {Object} An object of menu item anchor elements
+   */
+  _setUpMenu(menu) {
+    const menuObject = {};
+
+    // Create a real Array of all the focusable elements in the menu
+    const menuFocusables = Array.prototype.slice.call(
+      menu.querySelectorAll(this.focusableElements)
+    );
+
+    // Create a property to hold an array of all focusables
+    menuObject.all = menuFocusables;
+
+    // Create a property with a reference to the first focusable
+    menuObject.first = menuFocusables[0];
+
+    // Create a property with a reference to the last focusable
+    menuObject.last = menuFocusables[menuFocusables.length - 1];
+
+    return menuObject;
+  }
+
+  _handleClick(event) {
+    const dropdown = event.target.closest(this.dropdownAttribute);
+
+    // If click didn't come from within dropdown component, close all open menus
+    if (!dropdown) {
+      this.closeMenu();
+      return;
+    }
+
+    const toggleButton = dropdown.querySelector(this.toggleAttribute);
+    const menuList = dropdown.querySelector(this.menuAttribute);
+
+    // Delegate event to only this instance of the dropdown
+    if (dropdown === this.element) {
+      const menuTarget = event.target.closest(this.menuAttribute);
+
+      // Use this boolean on the event object in place of stopPropagation()
+      if (menuTarget && menuTarget !== null) {
+        event.clickedWithinMenu = true;
+      }
+
+      if (!toggleButton || toggleButton.getAttribute('aria-expanded') === 'true') {
+        /**
+         * Close the menu if there is a click outside of the menu, but within 
+         * the dropdown component
+         */
+        if (!event.clickedWithinMenu) {
+          this.closeMenu(toggleButton, menuList);
+        }
+
+        return;
+      }
+  
+      this.openMenu(toggleButton, menuList);
+    } else {
+      this.closeMenu();
+    }
+  }
+
+  _handleKeydown(event) {
+    const dropdown = event.target.closest(this.dropdownAttribute);
+
+    if (!dropdown) return;
+
+    const toggleButton = dropdown.querySelector(this.toggleAttribute);
```

Comment:
Same as above with these reassignments. You can reference `this.toggleElement`, etc. instead.